python/openai/README.md

```diff
@@ -0,0 +1,148 @@
+# OpenAI-Compatible Frontend for Triton Inference Server
+
+## Pre-requisites
+
+1. Docker + NVIDIA Container Runtime
+2. A correctly configured `HF_TOKEN` for access to HuggingFace models.
+    - The current examples and testing primarily use the
+      [`meta-llama/Meta-Llama-3-8B-Instruct`](https://huggingface.co/meta-llama/Meta-Llama-3-8B-Instruct)
+      model, but you can manually bring your own models and adjust accordingly.
+
+## VLLM
+
+1. Build and launch the container:
+  - Mounts the openai source files to `/workspace` for simplicity, later on these will be shipped in the container.
+  - Mounts the `~/.huggingface/cache` for re-use of downloaded models across runs, containers, etc.
+  - Sets the [`HF_TOKEN`](https://huggingface.co/docs/huggingface_hub/en/package_reference/environment_variables#hftoken) environment variable to
+    access gated models, make sure this is set in your local environment if needed.
+
+```bash
+docker build -t tritonserver-openai-vllm -f docker/Dockerfile.vllm .
+
+docker run -it --net=host --gpus all --rm \
+  -v ${PWD}:/workspace \
+  -v ${HOME}/.cache/huggingface:/root/.cache/huggingface \
+  -e HF_TOKEN \
+  -w /workspace \
+  tritonserver-openai-vllm
+```
+
+2. Launch the OpenAI server:
+```bash
+# NOTE: Adjust the --tokenizer based on the model being used
+python3 openai/main.py --model-repository openai/tests/vllm_models/ --tokenizer meta-llama/Meta-Llama-3.1-8B-Instruct
+```
+
+3. Send a `/v1/chat/completions` request:
+  - Note the use of `jq` is optional, but provides a nicely formatted output for JSON responses.
+```bash
+MODEL="llama-3.1-8b-instruct"
+curl -s http://localhost:8000/v1/chat/completions -H 'Content-Type: application/json' -d '{
+  "model": "'${MODEL}'",
+  "messages": [{"role": "user", "content": "Say this is a test!"}]
+}' | jq
+```
+
+4. Send a `/v1/completions` request:
+  - Note the use of `jq` is optional, but provides a nicely formatted output for JSON responses.
+```bash
+MODEL="llama-3.1-8b-instruct"
+curl -s http://localhost:8000/v1/completions -H 'Content-Type: application/json' -d '{
+  "model": "'${MODEL}'",
+  "prompt": "Machine learning is"
+}' | jq
```

Comment:
nit: can be done later, I think expected answer can help users

Reply:
Will improve with expected outputs in a follow-up PR focused on docs DLIS-7179